### PR TITLE
Fix memory issues

### PR DESCRIPTION
**Pull request by AnastasiaSliusar**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -51,14 +66,17 @@ export const initUntarJS = async (): Promise<IUnpackJSAPI> => {
...
+      if (errorMessagePtr !== 0) {
+        const errorMessage = wasmModule.UTF8ToString(errorMessagePtr);
+        console.error('Extraction failed:', errorMessage);
+      } else {
+        console.error('Extraction failed with unknown error.');
+      }
+      wasmModule._free(inputPtr);
+      wasmModule._free(fileCountPtr);
+      wasmModule._free_extracted_archive(resultPtr);
       return {};
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply src/index.ts:69]: Let's resolve that separately from this PR!

File: src/index.ts [hunk 2]
```diff
@@ -96,15 +103,21 @@ export const initUntarJS = async (): Promise<IUnpackJSAPI> => {
         dataSize
       );
 
-      files[filename] = fileData;
-    }
+      const fileDataCopy = new Uint8Array(Array.from(fileData));
 
+      files[filename] = fileDataCopy;
+    }
+    let newFiles = { ...files };
```
Comment (martinRenou) [inline src/index.ts:110]: Since we are creating the `files` dictionary, this copy of `files` here should have no impact but reduce performances?

Comment (AnastasiaSliusar) [reply src/index.ts]: Fixed

**Review by martinRenou — APPROVED** [review]
Thanks! I'll release that tomorrow

**Pull request merged**